### ISSUE TITLE
Training accuracy when training with triplet loss

### ISSUE THREAD

**Issue by ba305** [KaiyangZhou/deep-person-reid#189]
Hi @KaiyangZhou, I have recently been training some triplet loss models, and am achieving good results on the test set. However, I have noticed that during training, the training accuracies are extremely low (when using triplet loss only), even though the performance on the test set is good. 

However, when I train using --weight_x 0.5 and --weight_t 0.5 (i.e., using both triplet and softmax loss), the accuracies look normal in the output. Do you know why that is? It makes it a bit difficult to track the accuracy during training. I have also attached a picture to show you what I mean. You can reproduce this with the following command:

python main.py \
--root /PATH/TO/DATA \
--app image \
--loss triplet \
--train-sampler RandomIdentitySampler \
-s market1501 \
-a resnet50 \
--optim adam \
--lr 0.0003 \
--max-epoch 220 \
--stepsize 150 200 \
--batch-size 120 \
--save-dir /PATH/TO/SAVE \
--gpu-devices 0 \
--weight-t 1 \
--weight-x 0 \
--eval-freq 20 \
--random-erase

![image](https://user-images.githubusercontent.com/35350330/59386550-e9182400-8d34-11e9-939b-a3752bbac5ae.png)

Thank you!

**Comment by KaiyangZhou** [comment]
acc means the classification accuracy, which only makes sense when you enable softmax

**Comment by ba305** [comment]
Ok, thanks. I was under the impression that even when using triplet loss, the model still returns the classification output (see [here](https://github.com/KaiyangZhou/deep-person-reid/blob/e3c326677a93a79c991f3801089aa4b347704ea2/torchreid/models/resnet.py#L280) for example), in which case I would have thought that the training classification would still work. Do you know why it does not? 

This is not super important, so please feel free to ignore this if it is any trouble. thanks again for your help.

**Comment by KaiyangZhou** [comment]
when `weight_x` is 0, the classification layer receives zero gradients so it won't be updated
